### DISCUSSION TITLE
[Closed] Rename IsListNotNullOrEmptyConverter to IsListNotNullOrNotEmptyConverter

### DISCUSSION THREAD

**Discussion by softlion** [CommunityToolkit/Maui#588] [General]
The original name is confusing.

**Comment by VladislavAntonyuk** [comment]
It would be a breaking change. We should probably create a new converter and obsolete the previous one. It can be a part of .NET 7 release.

**Comment by VladislavAntonyuk** [comment]
@brminnick so to summarize, we are not going to rename it. Read converter name as IsListNot_NullOrEmpty_Converter, similar to string.IsNullOrEmpty

**Reply by softlion** [reply]
So you agree there is an issue with the naming, and that's it's not understandable until you dive into the doc. That's not the spirit of maui.
What's holding you back is only the breaking change, a "technical" issue.

**Reply by softlion** [reply]
Look at .net, is there a "IsNotNullOrEmpty" function ? I suppose you now know why  ;)

**Reply by TheCodeTraveler** [reply]
> So you agree there is an issue with the naming, and that's it's not understandable until you dive into the doc

No. We agree that a small percentage of developers may interpret the name as you have. But renaming it would still cause a small number of people to misinterpret the naming, just in the opposite direction. 

Changing the naming doesn’t solve the problem. 

However, ensuring our docs and XML comments (intellisense) explicitly state the expected results **does** prevent this confusion.

**Reply by softlion** [reply]
Ok I'm opening a poll on my fb maui group and will post the result here, so we'll know if i'm in a stupid 0.1% group of developers.

**Comment by TheCodeTraveler** [comment]
Here is the time-stamped link to our conversation on this topic during the December 2022 Monthly Standup: https://youtu.be/tqQhW104UKI?t=2881

**Reply by softlion** [reply]
After less than 1h / 6 votes 👍
![image](https://user-images.githubusercontent.com/190756/205454371-63359333-55a4-4677-925c-c13c31b6b3a2.png)

i'll share the link in 2 hours.

**Reply by TheCodeTraveler** [reply]
Did you include links to [its documentation](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/converters/is-list-not-null-or-empty-converter) and the [XML (intellisense) comments](https://github.com/CommunityToolkit/Maui/blob/23d9c54034a08a33e36a9ca8c5b9bdcd86881065/src/CommunityToolkit.Maui/Converters/IsListNotNullOrEmptyConverter.shared.cs#L7)?

Did you also include the fact that we have a converter named `IsListNullOrEmptyConverter`? And include a [link to its documentation](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/converters/is-list-null-or-empty-converter) and its [XML (intellisense) comments](https://github.com/CommunityToolkit/Maui/blob/23d9c54034a08a33e36a9ca8c5b9bdcd86881065/src/CommunityToolkit.Maui/Converters/IsListNullOrEmptyConverter.shared.cs#L7)?

[This Discussion](https://github.com/CommunityToolkit/Maui/discussions/588) and this poll is a bit misleading out of context.

**Comment by TheCodeTraveler** [comment]
Thank you softlion.

As discussed in various threads of [this Discussion](https://github.com/CommunityToolkit/Maui/discussions/588) and in our [December 2022 Monthly Standup](https://youtu.be/tqQhW104UKI?t=2881), we have come to the decision that we will not be changing the name of this Converter.

I understand you disagree with our naming and that's fine. Naming is hard. The English language allows for multiple interpretations which makes things difficult. And interpreting boolean logic from an English word/sentence only compounds this difficulty.

The good news is that we have documentation to provide context around the naming to avoid confusion. Out of context, yes this name may be confusing. With context, it should be clear. If our Docs + XML (intellisense) can be improved, please open a PR to help us fix that.

I hope you understand that we (the core maintainers) are a group of 7 volunteers who contribute our nights and weekends to this project. We do not all work for Microsoft. And none of us get paid to work on this project.

We have limited time to devote to every new feature, every new bug and every discussion. 

I hope you understand that we have spent hours of our time discussing [this Discussion](https://github.com/CommunityToolkit/Maui/discussions/588,) taking time away from us fixing high-priority bugs and implementing New Features. 

We are not ignoring you. We are engaging with you to help you understand why we will not be changing the name of this Converter.

I am closing this Discussion. I hope you understand our decision and I hope that we've given you enough context via [this Discussion](https://github.com/CommunityToolkit/Maui/discussions/588) and our [monthly standup](https://youtu.be/tqQhW104UKI?t=2881) to help you understand our decision.

**Comment by CliffAgius** [comment]
Closed as answered
